src/scripts/main.js

```diff
@@ -1,3 +1,525 @@
+      target.innerText = initialValue;
+      notification(pushInvalidNameText);
+      input.remove();
+
+      return;
+    }
+
+    if (isAge && (+input.value < 18 || +input.value > 90)) {
+      target.innerText = initialValue;
+      notification(pushInvalidAge);
+      input.remove();
+
+      return;
+    }
+
+    if (isOffice && !officeCities.includes(input.value)) {
+      target.innerText = initialValue;
+      notification(pushCellWrongOffice);
+      input.remove();
+
+      return;
+    }
+
+    target.innerText = input.value;
+    notification(pushCellSaved);
+    input.remove();
+  };
+
+  input.onkeydown = (keyboardEvent) => {
+    const isEnter = keyboardEvent.code === 'Enter';
+
+    if (!input.value && isEnter) {
+      target.innerText = initialValue;
+      notification(pushEmptyCell);
+      input.remove();
+
+      return;
+    }
+
+    if ((isAge || isSalary) && !input.value.match(/[0-9]/) && isEnter) {
+      target.innerText = initialValue;
+      notification(pushCellNotNumber);
+      input.remove();
+
+      return;
+    }
+
+    if (isStringField && input.value.match(/[0-9]/) && isEnter) {
+      target.innerText = initialValue;
+      notification(pushCellNotString);
+      input.remove();
+
+      return;
+    }
+
+    if (isSalary && input.value.match(/[0-9]/) && isEnter) {
+      target.innerText = getSalaryStr(input.value);
+      notification(pushCellSaved);
+      input.remove();
+
+      return;
+    }
+
+    if (isName && input.value.length < 4 && isEnter) {
+      target.innerText = initialValue;
+      notification(pushInvalidNameLength);
+      input.remove();
+
+      return;
+    }
+
+    if (isName && !input.value.match(/^[a-zA-Z ]*$/g) && isEnter) {
+      target.innerText = initialValue;
+      notification(pushInvalidNameText);
+      input.remove();
+
+      return;
+    }
+
+    if (isAge && (+input.value < 18 || +input.value > 90) && isEnter) {
+      target.innerText = initialValue;
+      notification(pushInvalidAge);
+      input.remove();
+
+      return;
+    }
+
+    if (isOffice && !officeCities.includes(input.value) && isEnter) {
+      target.innerText = initialValue;
+      notification(pushCellWrongOffice);
+      input.remove();
+
+      return;
+    }
+
+    if (isEnter) {
+      target.innerText = input.value;
+      notification(pushCellSaved);
+      input.remove();
+    }
+  };
+});
+
+function getWorkerInfo(workerData) {
+  return workerData.map(el => el.innerText);
+}
+
+function sort(workers, value, count) {
+  if (count % 2 !== 0) {
+    return workers.sort(
+      ({ [value]: a }, { [value]: b }) => typeof a === 'string'
+        ? a.localeCompare(b)
+        : a - b
+    );
+  }
+
+  return workers.sort(
+    ({ [value]: a }, { [value]: b }) => typeof a === 'string'
+      ? b.localeCompare(a)
+      : b - a
+  );
+}
+
+function checkClick() {
+  let action = '';
+  let counter = 0;
+
+  return function(value) {
+    if (!action || action === value) {
+      counter++;
```

Comment:
For boolean variable you can switch value like this `youVariable = !youVariable`